src/overlays/actors/ovl_Obj_Toge/z_obj_toge.c

```diff
@@ -32,52 +35,277 @@ const ActorInit Obj_Toge_InitVars = {
     (ActorFunc)ObjToge_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_809A4CB0 = {
-    { COLTYPE_METAL, AT_NONE, AC_ON | AC_HARD | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x01C37BB6, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_METAL,
+        AT_NONE,
+        AC_ON | AC_HARD | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x01C37BB6, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 30, 20, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_809A4D14[] = {
+f32 D_809A4CDC[] = { 2.0f, 4.0f, 6.0f, 8.0f, 10.0f, 12.0f, 14.0f, 16.0f };
+
+s16 D_809A4CFC[] = { 0, 10, 20, 30, 40, 50, 60, 70 };
+
+f32 D_809A4D0C[] = { 1.0f, 2.0f };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32_DIV1000(terminalVelocity, 0, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 150, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 150, ICHAIN_STOP),
 };
 
-#endif
+void func_809A41C0(ObjToge* this, GlobalContext* globalCtx) {
+    Vec3f sp1C = this->actor.world.pos;
+
+    sp1C.y += 10.0f;
+    sp1C.z += 10.0f;
+    func_800B2B44(globalCtx, &this->actor, &sp1C, 1.8f);
+
+    sp1C.x += 10.0f;
+    sp1C.z -= 20.0f;
+    func_800B2B44(globalCtx, &this->actor, &sp1C, 1.8f);
+
+    sp1C.x -= 20.0f;
+    func_800B2B44(globalCtx, &this->actor, &sp1C, 1.8f);
+}
+
+void func_809A42A0(ObjToge* this, GlobalContext* globalCtx) {
+    Vec3f sp54;
+    s32 i;
+    s16 phi_s0;
+
+    sp54.y = this->actor.world.pos.y + 15.0f;
+
+    for (i = 0, phi_s0 = 0; i < 5; i++, phi_s0 += 0x3333) {
+        sp54.x = (Math_SinS(phi_s0) * 15.0f) + this->actor.world.pos.x;
+        sp54.z = (Math_CosS(phi_s0) * 15.0f) + this->actor.world.pos.z;
+        func_800B2B44(globalCtx, &this->actor, &sp54, (Rand_ZeroOne() * 0.6f) + 2.8f);
+    }
+}
+
+void func_809A43A8(ObjToge* this, GlobalContext* globalCtx) {
+    if (!OBJTOGE_GET_4000(&this->actor)) {
+        func_809A41C0(this, globalCtx);
+    } else {
+        func_809A42A0(this, globalCtx);
+    }
+}
+
+s32 func_809A43EC(ObjToge* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    f32 temp_fv1 = player->actor.world.pos.x - this->unk_1B8;
```

Comment:
```suggestion
    Player* player = GET_PLAYER(globalCtx);

    f32 temp_fv1 = player->actor.world.pos.x - this->unk_1B8;
```

Reply:
Hmm, looks like temp_fv1 is being declared right there, so shouldn't need a space?

Reply:
Oh right, it's one of those functions where everything's a temp.